nuxeo-features/nuxeo-platform-oauth/src/main/java/org/nuxeo/ecm/platform/oauth2/request/AuthorizationRequest.java

```diff
@@ -82,21 +84,46 @@ public AuthorizationRequest(HttpServletRequest request) {
 
     public OAuth2Error checkError() {
         // Check mandatory fields
-        if (StringUtils.isBlank(responseType) || StringUtils.isBlank(clientId) || StringUtils.isBlank(redirectUri)) {
+        if (StringUtils.isBlank(responseType) || StringUtils.isBlank(clientId) || StringUtils.isBlank(redirectUri)
+                || !OAuth2Client.isRedirectURIValid(redirectUri)) {
             return OAuth2Error.INVALID_REQUEST;
         }
 
         // Check if client exists
+        ClientRegistry registry = Framework.getService(ClientRegistry.class);
+        OAuth2Client client = null;
         try {
-            ClientRegistry registry = Framework.getLocalService(ClientRegistry.class);
-            if (!registry.hasClient(clientId)) {
+            client = registry.getClient(clientId);
+            if (client == null || !client.isEnabled()) {
                 return OAuth2Error.UNAUTHORIZED_CLIENT;
             }
         } catch (DirectoryException e) {
             log.warn(e, e);
             return OAuth2Error.SERVER_ERROR;
         }
 
+        // Check that the client has a name since it is now a required field but it might be empty for an old client
+        String clientName = client.getName();
+        if (StringUtils.isBlank(clientName)) {
+            log.error(String.format(
+                    "No name set for OAuth2 client %s. It is a required field, please make sure you update this OAuth2 client.",
+                    client));
+        }
```

Comment:
Please add a comment after the `log.error` saying that we don't return an error for backward compatibility because we want old clients to keep working. You can then simplify the comment a few lines above. I think it's important that the comment be here after the `log.error` because that's where the decision is taken in the program flow and in other cases we always do `return OAuth2Error.XYZ` so not having a `return` is surprising.
Same for blank `clientRequestURI` below.